## SVM model:

In [3]:
import pandas as pd

# Load the CSV files
features_path = 'features.csv'
targets_path = 'targets.csv'

# Reading the CSV files
features_df = pd.read_csv(features_path)
targets_df = pd.read_csv(targets_path)

# # Display the first few rows of each dataframe
# features_head = features_df.head()
# targets_head = targets_df.head()

# features_head, targets_head


In [4]:
from sklearn import svm
from sklearn.model_selection import cross_val_score
from sklearn.preprocessing import StandardScaler
from sklearn.pipeline import make_pipeline


In [5]:
# SVM Classifier with cubic kernel
svm_clf = svm.SVC(kernel='poly', degree=3, C=1, decision_function_shape='ovo')

# Create a pipeline that automatically scales the data before feeding it to the SVM
pipeline = make_pipeline(StandardScaler(), svm_clf)

# Extract features and target variables
X = features_df.values
y = targets_df['Class'].values

# Perform 10-fold cross-validation
cv_scores = cross_val_score(pipeline, X, y, cv=10)

cv_scores.mean(), cv_scores.std()


(0.8832737761924646, 0.12565166827974095)

Below code takes several minutes, be careful !

In [6]:
# from sklearn.model_selection import GridSearchCV

# # Defining the parameter grid
# param_grid = {
#     'svc__gamma': ['scale', 'auto', 1e-2, 1e-3, 1e-4],
#     'svc__tol': [1e-3, 1e-4, 1e-5]
# }

# # Create a GridSearchCV object with 10-fold cross-validation
# grid_search = GridSearchCV(pipeline, param_grid, cv=10, scoring='accuracy')

# # Perform grid search
# grid_search.fit(X, y)

# # Best parameters and best score
# best_params = grid_search.best_params_
# best_score = grid_search.best_score_

# best_params, best_score

({'svc__gamma': 'scale', 'svc__tol': 0.001}, 0.8832737761924646)

({'svc__gamma': 'scale', 'svc__tol': 0.001}, 0.8832737761924646) was the result of the above code.